### PR TITLE
Add/analytics components

### PR DESCRIPTION
**Pull request by mustran**
## OVERVIEW

_Give a brief description of what this PR does._

## WHERE SHOULD THE REVIEWER START?

_e.g. `/src/elements/SomeComponent.js`_

## HOW CAN THIS BE MANUALLY TESTED?

_List steps to test this locally._

## ANY NEW DEPENDENCIES ADDED?

_List any new dependencies added._

- [ ] Does it work in IE >= 11?
- [ ] _Does it work in other browsers?_

## SCREENSHOTS (if applicable)

_Does your change affect the UI? If so, please add a screenshot._

## CHECKLIST

_Be sure all items are_ ✅ _before submitting a PR for review._

- [ ] Verify the linters and tests pass: `yarn review`
- [ ] Verify you bumped the lib version according to [Semantic Versioning Standards](http://semver.org)
- [ ] Verify you updated the CHANGELOG
- [ ] Verify this branch is rebased with the latest master

### CODE REVIEW

File: src/analytics/index.tsx [hunk 1]
```diff
@@ -37,8 +37,8 @@ export interface AnalyticsProps {
 }
 
 export type AnalyticsHandlerFn = (event: {
-  event: 'click';
-  type: 'button';
+  event: 'click' | 'dropdown_change' | 'dropdown_select';
```
Comment (mirovladimitrovski) [inline src/analytics/index.tsx]: I would put all these possible values into enums:
```
enum AnalyticsEvents {
  CLICK = 'click',
  DROPDOWN_CHANGE = 'dropdown_change',
  DROPDOWN_SELECT = 'dropdown_select'
}
```

and update this line accordingly
```
event: AnalyticsEvents
```

**Review by mirovladimitrovski — DISMISSED** [review]

File: src/analytics/index.tsx [hunk 2]
```diff
@@ -56,10 +56,12 @@ export class AnalyticsTracker {
   };
 
   track = (event: {
-    event: 'click';
```
Comment (Jasminna) [inline src/analytics/index.tsx:59]: And you can extend `AnalyticsEvents` enum with additional possible values and update this line also.

**Review by Jasminna — CHANGES_REQUESTED** [review]

File: src/analytics/index.tsx (hunk 2 above)
Comment (mirovladimitrovski) [reply src/analytics/index.tsx:59]: Or simply narrow it down to the single event you need: `event: AnalyticsEvents.CLICK`

**Review by Jasminna — DISMISSED** [review]

**Review by Jasminna — DISMISSED** [review]

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -30,26 +36,69 @@ const Dropdown: React.FC<Props> = ({
...
+                  tracker.track({
+                    event: AnalyticsEvents.DROPDOWN_CHANGE,
+                    type: AnalyticsComponentType.DROPDOWN,
+                    tag: `dropdown_${snakeCase(e.currentTarget.options[selectedValue].text)}`,
+                    pages,
+                    merchantId,
+                    ip,
+                  });
+                }
+          }
```
Comment (mirovladimitrovski) [inline src/components/Dropdown/Dropdown.tsx]: Normally I would be against inline function props, but you're already inside a function with its own arguments and extracting this function into a separate const would mean defining it in a scope where `pages`, `tracker`, `merchantId` and `ip` don't exist, so it's fine leaving it like this.

But how about making the following little change and incorporate the `!tag` condition inside the event handler:

```suggestion
          onChange={(e) => {
            onDropdownChange?.(e);
    
            if (tag) {
              const selectedValue = e.currentTarget.selectedIndex;
              tracker.track({
                event: AnalyticsEvents.DROPDOWN_CHANGE,
                type: AnalyticsComponentType.DROPDOWN,
                tag: `dropdown_${snakeCase(e.currentTarget.options[selectedValue].text)}`,
                pages,
                merchantId,
                ip,
              });
            }
          }}
```

**Review by mirovladimitrovski — DISMISSED** [review]

**Review by mirovladimitrovski — APPROVED** [review]

**Pull request merged**